### DISCUSSION TITLE
MacOS flashing failed

### DISCUSSION THREAD

**Discussion by smadawerdna** [forkineye/ESPixelStick#738] [Q&A]
New here- I'm trying to flash a new ESPixelStick and am getting the error "unable to find firmware configuration file". I am launching from terminal. Screenshot attached. Any guidance appreciated. 
<img width="738" alt="Screenshot 2024-02-09 at 9 35 52 PM" src="https://github.com/forkineye/ESPixelStick/assets/159599423/599a1f7d-1909-438c-a1fc-7e0da39893f2">

**Comment by forkineye** [comment]
Change to the extracted directory and run it from there. It uses relative paths to load the files and needs to be ran from there.

**Reply by smadawerdna** [reply]
Maybe I am missing something, but this is the extracted directory. Should it be pointing somewhere else?

**Reply by forkineye** [reply]
> Maybe I am missing something, but this is the extracted directory. Should it be pointing somewhere else? 

Change to the extracted directory first so you can run 'java -jar ESPSFlashTool.jar'. You have to launch it from its directory.

**Reply by MartinMueller2003** [reply]
Also, If you are loading Beta 5 then you are way out of date. You should load the distribution from the latest build.

**Reply by smadawerdna** [reply]
I was able to get into the Flash Tool, however I now cannot find the correct serial port. I've installed (removed then reinstalled again) the CH340 driver but it will still not connect. I also verified the cable I am using is a data cable. 
<img width="539" alt="Screenshot 2024-02-10 at 4 32 50 PM" src="https://github.com/forkineye/ESPixelStick/assets/159599423/cdf1d1bd-925e-4822-817b-79d4689a491f">

**Comment by forkineye** [comment]
That is not the correct serial port. Either your USB cable is power only (no data) or you didn't allow the USB to Serial device when MacOS does the security pop-up. You should something similar to `cu.usberial-10 - USB 2.0-Serial` as shown in the screenshot. Also, you should already have the CH340 drivers.
<img width="1021" alt="image" src="https://github.com/forkineye/ESPixelStick/assets/4047923/59b2690a-8de1-4aa6-9c1f-b0891d99dc8d">

**Comment by robwmc** [comment]
I am having an issue also. I'm in the extracted directory to run 'java -jar ESPSFlashTool.jar' but every time I get this.

![Error1](https://github.com/forkineye/ESPixelStick/assets/10272147/93f37f83-ff63-492d-9323-7dc6e62b436a)

**Reply by robwmc** [reply]
Here is some food for thought. Nothing I did on my Mac ever worked, even after allowing the application to work in the Privacy and Security tab in System Preferences. I'm running Ventura 13.6.1 on Intel.

I tried it on a MacBook Pro running Ventura 13.6.1 on an Apple M2 chip. Same result.

I tried to use it on my PC, which is running Win11. I'm not in front of it so I can't tell you what update it has. 

As a last resort, I grabbed my daughter's Ailenware gaming PC running Win10 and was able to successfully flash it. I don't know if this is of any help but I thought I would pass along the particulars in case it meant anything. Very happy to begin the experimenting now!

**Comment by MartinMueller2003** [comment]
You might want to try the https://espixelstickwebflasher.from-ct.com:5000/
